domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/validation/casefield/CaseFieldEntityComplexFieldACLValidatorImplTest.java

```diff
@@ -124,6 +139,28 @@ public void shouldHaveValidationErrorWhenUserNotFound() {
         );
     }
 
+    @Test
+    @DisplayName("should fail when user has no valid access")
+    public void shouldHaveValidationErrorWhenUserHasNoValidAccess() {
+        given(caseFieldEntityValidationContext.getCaseTypeCaseRoles()).willReturn(newHashSet(ROLE_DEFENDANT));
+
+        complexACL0.setUserRole(caseRoleCreator);
+        complexACL0.setCreate(true);
+        caseField.addComplexFieldACL(complexACL0);
+        fieldACL.setCreate(true);
+        fieldACL.setUserRole(userRole);
+        caseField.addCaseFieldACL(fieldACL);
+
+        final ValidationResult result = validator.validate(caseField, caseFieldEntityValidationContext);
+
+        assertAll(
+            () -> assertThat(result.getValidationErrors().size(), is(2)),
+            () -> assertThat(result.getValidationErrors().get(0), instanceOf(CaseFieldEntityInvalidCaseRoleValidationError.class)),
+            () -> assertThat(result.getValidationErrors().get(0).getDefaultMessage(), is(
+                "Unknown case role 'case_type'. Please make sure it is declared in the list of supported case roles for the case type 'case_field'"))
```

Comment:
is this correct? why are we using `case_type` and `case_field`? the first one should be the case role and the second one the case type.